woocommerce/woocommerce-gateway-stripe
# Display notice on the empty state connect UI when the OAuth URL is empty

Fixes #3314 

## Changes proposed in this Pull Request:

In https://github.com/woocommerce/woocommerce-gateway-stripe/issues/3196 I removed the flow that enabled users to enter API keys manually if we fail to generate an OAuth URL. This currently leaves users in a dead end with no explanation. 

This PR attempts to improve that UX by displaying a notice.  

<p align="center">
<img width="500" alt="Screenshot 2024-07-26 at 1 09 49 PM" src="https://github.com/user-attachments/assets/70d8c612-e028-4220-95bc-3760d5465a27">
</br>
<sup>Notice shown to users when we failed to generate a OAuth  URL.</sup>
</p> 

## Testing instructions

1. Checkout this branch and run `npm run build`.
1. Backup your Stripe settings by changing the `woocommerce_stripe_settings` option in your database to `woocommerce_stripe_settings-backup`. Deleting it entirely would also work. 
2. Go to **WooCommerce > Settings > Payments > Stripe**. 
3. You should see the option to **Create or connect an account**.
4. Mimic a failure in generating the OAuth URL. 
   - Either change your site to not be served over https or just change [this logic](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/add/display-notice-on-empty-oauth-url/includes/connect/class-wc-stripe-connect.php#L46-L48) to make it pass mimicking that kind of failure.
5. The empty state connect flow should now display a notice.
   - Prior to this PR there was nothing displayed. See example [here](https://github.com/user-attachments/assets/f9ede0fc-71fa-4c05-9d27-31dd52736858). 

---


**Post merge**
